### DISCUSSION TITLE
Potentiometer and wemos mini

### DISCUSSION THREAD

**Discussion by mokusnz** [arendst/Tasmota#16587] [Q&A]
Hi all, please help 
my, how to use potentiometer on wemos mini with tasmota?
I have a physical analog potentiometer connected via A0, I can see it in home assistant (%). I can't write to the config. yaml ..., I also have pwm and relay output on wemos ...
Please help my ...
console:
07:57:57.400 CMD: rule
07:57:57.407 MQT: stat/digipot/RESULT = {"Rule1":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
07:57:57.623 MQT: tele/digipot/STATE = {"Time":"1970-01-02T07:57:57","Uptime":"1T07:41:51","UptimeSec":114111,"Heap":21,"SleepMode":"Dynamic","Sleep":10,"LoadAvg":99,"MqttCount":6,"POWER1":"ON","POWER2":"ON","Dimmer":56,"Fade":"OFF","Speed":1,"LedTable":"ON","Wifi":{"AP":1,"SSId":"TP-LINK","BSSId":"B8:69:F4:3C:47:BD","Channel":1,"Mode":"11n","RSSI":48,"Signal":-76,"LinkCount":2,"Downtime":"0T00:00:13"}}
07:57:57.639 MQT: tele/digipot/SENSOR = {"Time":"1970-01-02T07:57:57","ANALOG":{"Range":71}}

![HA pot](https://user-images.githubusercontent.com/61285374/191073339-46181851-69d8-4ada-9de7-7fd4a68c3c80.jpg)
![w tasmota12](https://user-images.githubusercontent.com/61285374/191073341-a13e0458-2eda-4b98-8713-bd01af5c79a5.jpg)

**Comment by barbudor** [comment]
Your analog input seems to work as it provides an "ANALOG" key in the sensor message
07:57:57.639 MQT: tele/digipot/SENSOR = {"Time":"1970-01-02T07:57:57","ANALOG":{"Range":71}}

What do you want more ?

**Comment by mokusnz** [comment]
I measure the output of the dimmer (0-3.3V) when I use a potentiometer, the output does not respond when I control via PC (tasmota, mouse) only then does the control work on the dimmer... I still feel that nothing needs to be written in config.yaml, can you do it confirm?

**Comment by mokusnz** [comment]
![HA pot2](https://user-images.githubusercontent.com/61285374/191312260-6e34db94-b2c0-425d-9031-da8f7973783a.jpg)

**Comment by mokusnz** [comment]
![HA pot3](https://user-images.githubusercontent.com/61285374/191336237-7f2954db-9294-4ac8-8816-f489a0394950.jpg)

**Comment by barbudor** [comment]
There is no link between analog input and dimmer
If you want the potentiometer to control the dimmer you need a rule like
Rule1 On analog#range do dimmer %value% endon
Rule1 1

However with such a rule it is the potentiometer that control the dimmer. If you send a dimmer command from mqtt, it will be overwritten by the potentiometer value immediately after.

To allow control by both you need a more complex rule to set the dimmer from the potentiometer only if the potentiometer change by a value. You can have a look at the rule docs for a rule that detect changes in values

However potentiometer is not the best to contro' à dimmer, you should consider a rotary switch

**Comment by mokusnz** [comment]
THX, THX, THX, working ..., THX ...

**Comment by mokusnz** [comment]
![rule5](https://user-images.githubusercontent.com/61285374/191343555-0a64cb59-e25b-440d-9504-9f723fba1a9b.jpg)

**Comment by mokusnz** [comment]
The second task, I need to have a potentiometer on wemos1 (tasmota), a dimmer on wemos2 (tasmota), is it possible to do this?

**Reply by barbudor** [reply]
Yes, instead of executing the command on the local Wemos1, you can publish a command to the wemos2 using MQTT
Assuming the topic of the 2nd wemos is wemos2, the rule becomes:
Rule1 On analog#range do publish cmnd/wemos2/dimmer %value% endon

**Reply by mokusnz** [reply]
Thanks for the info and support, THX...
